src/Payum/Paypal/ProHosted/Nvp/Action/Api/BaseApiAwareAction.php

```diff
@@ -0,0 +1,28 @@
+<?php
+namespace Payum\Paypal\ProHosted\Nvp\Action\Api;
+
+use Payum\Core\Action\ActionInterface;
+use Payum\Core\Action\GatewayAwareAction;
+use Payum\Core\ApiAwareInterface;
+use Payum\Core\Exception\UnsupportedApiException;
+use Payum\Paypal\ProHosted\Nvp\Api;
+
+abstract class BaseApiAwareAction extends GatewayAwareAction implements ActionInterface, ApiAwareInterface
```

Comment:
this class is not needed, you can use ApiAwareInterface plush ApiAwareTrait

and set the desired apiClass in constructor.